### PR TITLE
Feature/radix UI modal

### PR DESCRIPTION
**Pull request by bielsky0**
### Please check if the PR fulfills these requirements

- [x] The commit message follows our guidelines
- [x] Tests for the changes have been added (for bug fixes/features)
- [ ] Docs have been added / updated (for bug fixes / features)

### What kind of change does this PR introduce?

Migrate from styled components to radix ui

### CODE REVIEW

File: packages/webapp-libs/webapp-core/src/components/modal/modal.component.tsx [hunk 1]
```diff
@@ -1,55 +1,96 @@
...
-            </Container>
-          </ClickAwayListener>
-        </Overlay>
+import * as DialogPrimitive from '@radix-ui/react-dialog';
+import { X } from 'lucide-react';
+import { forwardRef } from 'react';
+
+import { cn } from '../../lib/utils';
+
+const Dialog = DialogPrimitive.Root;
```
Comment (mkleszcz) [inline packages/webapp-libs/webapp-core/src/components/modal/modal.component.tsx:7]: We should keep the file naming consistent with the component name so either change the file name or change the component name

**Review by mkleszcz — CHANGES_REQUESTED** [review]

File: packages/webapp-libs/webapp-core/src/components/dialog/dialog.component.tsx
Comment (mkleszcz) [inline packages/webapp-libs/webapp-core/src/components/dialog/dialog.component.tsx:1]: Move each component in separate file, please ;)

File: packages/webapp-libs/webapp-core/src/components/dialog/dialog.styles.ts
Comment (mkleszcz) [inline packages/webapp-libs/webapp-core/src/components/dialog/dialog.styles.ts:1]: Is this file used?

**Review by mkleszcz — CHANGES_REQUESTED** [review]

File: packages/webapp-libs/webapp-core/src/components/dialog/__tests__/dialog.component.spec.tsx [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+import { render } from '../../../tests/utils/rendering';
+
+describe('Dialog: Component', () => {
+  const content = 'content';
+  const header = 'header';
+
+  it('should be open', async () => {
+    render(
+      <Dialog open={true}>
+        <DialogContent> {content}</DialogContent>
```
Comment (mkleszcz) [inline packages/webapp-libs/webapp-core/src/components/dialog/__tests__/dialog.component.spec.tsx:14]: This space is a typo?;)

Comment (bielsky0) [reply packages/webapp-libs/webapp-core/src/components/dialog/__tests__/dialog.component.spec.tsx:14]: yep

**Pull request merged**